### PR TITLE
Enable annotation-based local/remote test runner selection for `file-watching`

### PR DESCRIPTION
**Pull request by hegyibalint**
The gist of this PR is not as bad as the title:
 - Test distribution is used to run tests faster, but there are exemptions (e.g., the `file-watching` project) where we cannot use test distribution (we have special file-system tests, which need a specially prepped machine).
 - Currently, we enable the feature (depending on the project) by passing an extra parameter: https://github.com/gradle/gradle/blob/e62c3e96f7c3917d141c8441740ef628b9249a0b/.teamcity/src/main/kotlin/configurations/FunctionalTest.kt#L36-L41
 - Our target is to leverage annotation matching, so it's not the CI's responsibility to manage this, but rather the test's. This would be beneficial, as it's much easier to control the behavior that way.

### CODE REVIEW

File: subprojects/file-watching/build.gradle.kts [hunk 1]
```diff
@@ -26,6 +26,7 @@ dependencies {
     testImplementation(testFixtures(project(":snapshots")))
 
     testImplementation(libs.commonsIo)
+    testImplementation(libs.gradleEnterpriseTestAnnotation)
```
Comment (wolfs) [inline subprojects/file-watching/build.gradle.kts:29]: Does it work as well if you add the dependency here: https://github.com/gradle/gradle/blob/fc502b86e2a0915d2e678c98ce3c89db75582494/build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts#L125-L138

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/BuildOperationsFileSystemWatchingIntegrationTest.groovy [hunk 2]
```diff
@@ -16,10 +16,12 @@
 
 package org.gradle.internal.watch
 
+import com.gradle.enterprise.testing.annotations.LocalOnly
 import org.gradle.integtests.fixtures.BuildOperationsFixture
 import org.gradle.internal.watch.vfs.BuildFinishedFileSystemWatchingBuildOperationType
 import org.gradle.internal.watch.vfs.BuildStartedFileSystemWatchingBuildOperationType
 
+@LocalOnly
```
Comment (wolfs) [inline subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/BuildOperationsFileSystemWatchingIntegrationTest.groovy:24]: In order for the annotation to take effect, we'd also need to configure test selection in a way it respects the annotation, right? Like described here: https://docs.gradle.com/enterprise/test-distribution/#gradle_executor_restrictions_annotation_matcher

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/NoDaemonFilesystemWatchingIntegrationTest.groovy [hunk 3]
```diff
@@ -16,13 +16,14 @@
 
 package org.gradle.internal.watch
 
-
```
Comment (wolfs) [inline subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/NoDaemonFilesystemWatchingIntegrationTest.groovy:19]: 💄 Extra newline

File: subprojects/file-watching/build.gradle.kts (hunk 1 above)
Comment (hegyibalint) [reply subprojects/file-watching/build.gradle.kts:29]: Heh... probably, was just looking exactly a place like this.

Comment (hegyibalint) [reply subprojects/file-watching/build.gradle.kts:29]: Moved up there, seems to be working

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/BuildOperationsFileSystemWatchingIntegrationTest.groovy (hunk 2 above)
Comment (hegyibalint) [reply subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/BuildOperationsFileSystemWatchingIntegrationTest.groovy:24]: Correct, added

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/NoDaemonFilesystemWatchingIntegrationTest.groovy (hunk 3 above)
Comment (hegyibalint) [reply subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/NoDaemonFilesystemWatchingIntegrationTest.groovy:19]: Interesting, this was not automatically the format IDEA followed?

Comment (hegyibalint) [reply subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/NoDaemonFilesystemWatchingIntegrationTest.groovy:19]: Now I'm conflicted... there is a single newline here, that should be the standard

Comment (wolfs) [reply subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/NoDaemonFilesystemWatchingIntegrationTest.groovy:19]: Ah, sorry I looked wrong. Removing that newline is good!

**Review by wolfs — APPROVED** [review]
LGTM! The next step would be to remove the special case from the JSON generation.

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Pull request merged**